java/flight/flight-core/src/main/java/org/apache/arrow/flight/grpc/MetadataAdapter.java

```diff
@@ -43,16 +44,35 @@ public String get(String key) {
     return this.metadata.get(Key.of(key, Metadata.ASCII_STRING_MARSHALLER));
   }
 
+  @Override
+  public byte[] getByte(String key) {
+    if (key.endsWith(Metadata.BINARY_HEADER_SUFFIX)) {
+      return this.metadata.get(Key.of(key, Metadata.BINARY_BYTE_MARSHALLER));
+    }
+    return get(key).getBytes();
+  }
+
   @Override
   public Iterable<String> getAll(String key) {
     return this.metadata.getAll(Key.of(key, Metadata.ASCII_STRING_MARSHALLER));
   }
 
+  @Override
+  public Iterable<byte[]> getAllByte(String key) {
```

Comment:
For consistency with `getByte`, this should also check the key and use the correct decoder instead of always deferring to getAll.

Reply:
done